tests/tests.hs

```diff
@@ -794,114 +796,114 @@ instance Arbitrary VersionNumber where
-$(derive makeArbitrary ''Locale)
-$(derive makeArbitrary ''MaxWordLength)
-$(derive makeArbitrary ''MinWordLength)
-$(derive makeArbitrary ''PhraseSlop)
-$(derive makeArbitrary ''MinDocFrequency)
-$(derive makeArbitrary ''MaxDocFrequency)
-$(derive makeArbitrary ''Regexp)
-$(derive makeArbitrary ''SimpleQueryStringQuery)
-$(derive makeArbitrary ''FieldOrFields)
-$(derive makeArbitrary ''SimpleQueryFlag)
-$(derive makeArbitrary ''RegexpQuery)
-$(derive makeArbitrary ''QueryStringQuery)
-$(derive makeArbitrary ''RangeQuery)
-$(derive makeArbitrary ''RangeValue)
-$(derive makeArbitrary ''PrefixQuery)
-$(derive makeArbitrary ''NestedQuery)
-$(derive makeArbitrary ''MoreLikeThisFieldQuery)
-$(derive makeArbitrary ''MoreLikeThisQuery)
-$(derive makeArbitrary ''IndicesQuery)
-$(derive makeArbitrary ''HasParentQuery)
-$(derive makeArbitrary ''HasChildQuery)
-$(derive makeArbitrary ''FuzzyQuery)
-$(derive makeArbitrary ''FuzzyLikeFieldQuery)
-$(derive makeArbitrary ''FuzzyLikeThisQuery)
-$(derive makeArbitrary ''FilteredQuery)
-$(derive makeArbitrary ''DisMaxQuery)
-$(derive makeArbitrary ''CommonTermsQuery)
-$(derive makeArbitrary ''DistanceRange)
-$(derive makeArbitrary ''MultiMatchQuery)
-$(derive makeArbitrary ''LessThanD)
-$(derive makeArbitrary ''LessThanEqD)
-$(derive makeArbitrary ''GreaterThanD)
-$(derive makeArbitrary ''GreaterThanEqD)
-$(derive makeArbitrary ''LessThan)
-$(derive makeArbitrary ''LessThanEq)
-$(derive makeArbitrary ''GreaterThan)
-$(derive makeArbitrary ''GreaterThanEq)
-$(derive makeArbitrary ''GeoPoint)
-$(derive makeArbitrary ''NullValue)
-$(derive makeArbitrary ''MinimumMatchHighLow)
-$(derive makeArbitrary ''CommonMinimumMatch)
-$(derive makeArbitrary ''BoostingQuery)
-$(derive makeArbitrary ''BoolQuery)
-$(derive makeArbitrary ''MatchQuery)
-$(derive makeArbitrary ''MultiMatchQueryType)
-$(derive makeArbitrary ''BooleanOperator)
-$(derive makeArbitrary ''ZeroTermsQuery)
-$(derive makeArbitrary ''MatchQueryType)
-$(derive makeArbitrary ''SearchAliasRouting)
-$(derive makeArbitrary ''ScoreType)
-$(derive makeArbitrary ''Distance)
-$(derive makeArbitrary ''DistanceUnit)
-$(derive makeArbitrary ''DistanceType)
-$(derive makeArbitrary ''OptimizeBbox)
-$(derive makeArbitrary ''GeoBoundingBoxConstraint)
-$(derive makeArbitrary ''GeoFilterType)
-$(derive makeArbitrary ''GeoBoundingBox)
-$(derive makeArbitrary ''LatLon)
-$(derive makeArbitrary ''RangeExecution)
-$(derive makeArbitrary ''RegexpFlag)
-$(derive makeArbitrary ''BoolMatch)
-$(derive makeArbitrary ''Term)
-$(derive makeArbitrary ''IndexSettings)
-$(derive makeArbitrary ''UpdatableIndexSetting)
-$(derive makeArbitrary ''Bytes)
-$(derive makeArbitrary ''AllocationPolicy)
-$(derive makeArbitrary ''InitialShardCount)
-$(derive makeArbitrary ''FSType)
-$(derive makeArbitrary ''CompoundFormat)
-$(derive makeArbitrary ''FsSnapshotRepo)
-$(derive makeArbitrary ''SnapshotRepoName)
+instance Arbitrary IndexName where arbitrary = sopArbitrary; shrink = genericShrink
```

Comment:
When `QuickCheck` gets generic derivation, we'll be able to drop the `where` statements